diff --git a/Gaudi/cmake/GaudiProjectConfig.cmake b/Gaudi/cmake/GaudiProjectConfig.cmake
index bf8ae091..08aade3b 100644
--- a/Gaudi/cmake/GaudiProjectConfig.cmake
+++ b/Gaudi/cmake/GaudiProjectConfig.cmake
@@ -407,7 +407,7 @@ macro(gaudi_project project version)
       # Avoid interference from user environment
       unset(ENV{GIT_DIR})
       unset(ENV{GIT_WORK_TREE})
-      execute_process(COMMAND git clone https://gitlab.cern.ch/gaudi/xenv.git ${CMAKE_BINARY_DIR}/contrib/xenv)
+      execute_process(COMMAND git clone https://gitlab.cern.ch/hgraslan/xenv.git ${CMAKE_BINARY_DIR}/contrib/xenv)
     endif()
     # I'd like to generate the script with executable permission, but I only
     # found this workaround: https://stackoverflow.com/a/45515418
